src/expr/src/vector_op/array_length.rs

```diff
@@ -126,8 +127,8 @@ fn array_length(array: ListRef<'_>) -> Result<i64, ExprError> {
 /// statement error
 /// select array_length(array[null, array[2]], 2);
 /// ```
-#[function("array_length(list, int32) -> int64")]
-fn array_length_d(array: ListRef<'_>, d: i32) -> Result<Option<i64>, ExprError> {
+#[function("array_length(list, int32) -> int32")]
+fn array_length_of_dim(array: ListRef<'_>, d: i32) -> Result<Option<i32>, ExprError> {
```

Comment:
This variant is just added by #10197 and we do not need to worry about backward compatibly.